File: unijava/src/org/unified/uni/declaration/DeclarationAttribute.java
package org.unified.uni.declaration;

/**
 * Uni Declaration Attribute
 * <p>
 * Created by Michael Vlasov on 19.04.2018.
 * <p>
 * Copyright (c) 2018 Michael Vlasov
 */
public final class DeclarationAttribute {
	public final String name;
	public final Object value;


	public DeclarationAttribute(String name, Object value) {
		this.name = name;
		this.value = value;
	}


	public static final DeclarationAttribute[] emptyArray = new DeclarationAttribute[0];
}

File: uniswift/Uni/Uni/Declaration/DeclarationAttribute.swift

//  UnElement.swift
//  Uni
//
//  Created by Michael Vlasov on 27.04.2018.
//  Copyright © 2018 Unified. All rights reserved.
//

import Foundation

public struct DeclarationAttribute {
	public enum Value {
		case missing
		case string(String)
		case array([String])
	}
	
	public let name: String
	public let value: Value
	
	init(name: String, value: Value) {
		self.name = name
		self.value = value
	}
}
